x/capability/abci.go

```diff
@@ -0,0 +1,21 @@
+package capability
+
+import (
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/capability/keeper"
+	"github.com/cosmos/cosmos-sdk/x/capability/types"
+)
+
+// BeginBlocker will call InitMemStore to initialize the memory stores in the case
+// that this is the first time the node is executing a block since restarting (wiping memory).
+// In this case, the BeginBlocker method will reinitialize the memory stores locally, so that subsequent
+// capability transactions will pass.
+// Otherwise BeginBlocker performs a no-op.
+func BeginBlocker(ctx sdk.Context, k keeper.Keeper) {
+	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyBeginBlocker)
+
+	k.InitMemStore(ctx)
```

Comment:
This will be used if node is restarting without genesis restart (ie. regular restart, upgrade with upgrade module, statesync)

Reply:
`BeginBlocker` will be called on genesis I think (i.e. the 1st block).

Reply:
Yes but the initialization logic won't get rerun because the initialization flag in memory is already set

Reply:
That is my understanding, please confirm @alexanderbez

Reply:
Correct, state-synced nodes will NOT run `InitGenesis` and will not run `BeginBlock` until the first block after the state-sync height.